Determine whether net/http needs to be removed in go lambdas

If the only usage of this import in code is the listener then we get 

```
#11 134.4 ./main.go:9:2: "net/http" imported and not used
```
during the dockerbuild


also remove the / route on our go mega app when this is complete

--- a/pkg/lang/golang/plugin_expose.go
+++ b/pkg/lang/golang/plugin_expose.go
@@ -1,6 +1,7 @@
 package golang
 
 import (
+	"fmt"
 	"path"
 	"strings"
 
@@ -183,6 +184,11 @@ func (h *restAPIHandler) handleFile(f *core.SourceFile) (*core.SourceFile, error
 			return nil, core.NewCompilerError(f, capNode, err)
 		}
 
+		err = h.removeNetHttpImport(f)
+		if err != nil {
+			return nil, core.NewCompilerError(f, capNode, err)
+		}
+
 		router, err := h.findChiRouterDefinition(f, routerName)
 		if err != nil {
 			return nil, core.NewCompilerError(f, capNode, err)
@@ -237,6 +243,37 @@ func (h *restAPIHandler) handleFile(f *core.SourceFile) (*core.SourceFile, error
 		}
 	}
 	return f, nil
+}
+
+func (h *restAPIHandler) removeNetHttpImport(f *core.SourceFile) error {
+	h.log.Info(fmt.Sprintf("searching for http imports in %s", f.Path()))
+	netHttpImportName := "http"
+	netHttpImport := GetNamedImportInFile(f, netHttpImportName)
+	if netHttpImport.Alias != "" {
+		netHttpImportName = netHttpImport.Alias
+	}
+
+	nextMatch := doQuery(f.Tree().RootNode(), fmt.Sprintf("[((package_identifier)@id (#match? @id \"%s\")) ((identifier)@id (#match? @id \"%s\"))]", netHttpImportName, netHttpImportName))
+
+	httpUsed := false
+	for {
+		_, found := nextMatch()
+		if !found {
+			break
+		}
+		if found {
+			httpUsed = true
+			break
+		}
+	}
+
+	if !httpUsed {
+		err := UpdateImportsInFile(f, []Import{}, []Import{{Package: "net/http"}})
+		if err != nil {
+			return errors.Wrap(err, "error updating imports")
+		}
+	}
+	return nil
 }
 
 func (h *restAPIHandler) findChiRouterDefinition(f *core.SourceFile, appName string) (chiRouterDefResult, error) {

--- a/pkg/lang/golang/plugin_expose_test.go
+++ b/pkg/lang/golang/plugin_expose_test.go
@@ -115,6 +115,91 @@ func Test_findChiRouterDefinition(t *testing.T) {
 	}
 }
 
+func Test_removeNetHttpImport(t *testing.T) {
+	tests := []struct {
+		name   string
+		source string
+		want   string
+	}{
+		{
+			name: "simple identifier",
+			source: `package test
+import (
+	"fmt"
+	"net/http"
+)
+
+http.ListenAndServe(":3000", r)
+			`,
+			want: `package test
+import (
+	"fmt"
+	"net/http"
+)
+
+http.ListenAndServe(":3000", r)
+			`,
+		},
+		{
+			name: "simple package identifier",
+			source: `package test
+import (
+	"fmt"
+	"net/http"
+)
+
+r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("Hello!"))
+})
+			`,
+			want: `package test
+import (
+	"fmt"
+	"net/http"
+)
+
+r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("Hello!"))
+})
+			`,
+		},
+		{
+			name: "should remove import",
+			source: `package test
+import (
+	"fmt"
+	"net/http"
+)
+
+random.ListenAndServe(":3000", r)
+			`,
+			want: `package test
+
+import (
+	"fmt"
+)
+
+
+random.ListenAndServe(":3000", r)
+			`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+
+			f, err := core.NewSourceFile("", strings.NewReader(tt.source), Language)
+			if !assert.NoError(err) {
+				return
+			}
+			err = testRestAPIHandler.removeNetHttpImport(f)
+
+			assert.NoError(err)
+			assert.Equal(tt.want, string(f.Program()))
+		})
+	}
+}
+
 func Test_findImports(t *testing.T) {
 	tests := []struct {
 		name   string

